### ISSUE TITLE
strtotime's possible false return isn't evaluated as parameter to another function?

### ISSUE THREAD

**Issue by jaydiablo** [vimeo/psalm#796]
This code:

```php
<?php

echo date('Y-m-d', strtotime(''));
```

Produces no errors in psalm: https://getpsalm.org/r/23844efe46

But strtotime in this situation returns `false`.   https://3v4l.org/mRkuP

This code:

```php
<?php

echo date('Y-m-d', false);
```

Does throw an error: https://getpsalm.org/r/70c49b85a2

**Comment by weirdan** [comment]
It's likely for the same reason as explained here: https://github.com/vimeo/psalm/issues/785#issuecomment-393475313

**Comment by jaydiablo** [comment]
Ah, interesting, this one seems like it should be caught by psalm, IMO, since invalid input to `strtotime` is hard to test for without using `strtotime`.

Maybe should be added as an exception, like `strpos` was?

**Comment by jaydiablo** [comment]
Can psalm see if strict types are enabled?

If so, perhaps it would be nice if the `@psalm-ignore-falsable-return` functionality is disabled in that case? (or could be behind a config option that developers can dis/enable?)

This particular code throws a Type Error when strict types are enabled, and it would be nice to be able to use psalm to help detect those cases.

**Comment by muglug** [comment]
`@psalm-ignore-falsable-return` is used where I decided that 99.9% of the time it's used in peoples' code, the function would not return `false`. Same with `preg_replace` (which returns `false` if the input contains malformed characters).

You can always stub those functions out:

```php
/** @return string|false */
function strtotime(string $time, int $now = -1) {}
```

and reference that stub in your config.

Otherwise I could also add a config flag, if you want the functionality more broadly.
